.github/workflows/build.yaml

```diff
@@ -21,6 +21,8 @@ on:
 
 jobs:
   detect:
+    permissions:
+      content: read
```

Comment:
nit: It is also possible to set this at the highest level, and then just override the permissions on jobs that require than `content: read`.